crates/bevy_ecs/src/archetype.rs

```diff
@@ -510,7 +515,7 @@ impl Archetypes {
             .archetype_ids
             .entry(archetype_identity)
             .or_insert_with(move || {
-                let id = ArchetypeId(archetypes.len());
+                let id = ArchetypeId::new(archetypes.len());
```

Comment:
Should we have a better panic message here? It's possible that this would fail somewhere else I guess